src/plugin-hooks/useSortBy.js

```diff
@@ -161,62 +161,70 @@ function useMain(instance) {
   }
 
   // Add the getSortByToggleProps method to columns and headers
-  flatHeaders.forEach(column => {
-    const { accessor, disableSorting: columnDisableSorting, id } = column
-
-    const canSort = accessor
-      ? getFirstDefined(
-          columnDisableSorting === true ? false : undefined,
-          disableSorting === true ? false : undefined,
-          true
-        )
-      : false
-
-    column.canSort = canSort
-
-    if (column.canSort) {
-      column.toggleSortBy = (desc, multi) =>
-        toggleSortBy(column.id, desc, multi)
-
-      column.clearSorting = () => {
-        return setState(old => {
-          const { sortBy } = old
-          const newSortBy = sortBy.filter(d => d.id !== column.id)
-          return {
-            ...old,
-            sortBy: newSortBy,
-          }
-        }, actions.sortByChange)
+  flatHeaders.forEach(
+    // avoid sharing instance in contexts of other functions
+    (instance => column => {
+      const { accessor, disableSorting: columnDisableSorting, id } = column
```

Comment:
I haven't thought of a better way to handle this. Wrapping the function seems to prevent memory leak.